File: sources/gui/Ggio.BikeSherpa.Frontend/couriers/models/Courier.ts
import * as Crypto from "expo-crypto";
import InputCourier from "./InputCourier";
import Storable from "@/models/Storable";
import { Address } from "@/models/Address";
import { HateoasLinks, Link } from "@/models/HateoasLink";
import { z } from "zod";
import { schemas } from "@/infra/openAPI/client";

export default class Courier extends InputCourier implements Storable, HateoasLinks {
    // Storable
    public readonly id: string;
    public createdAt?: string;
    public updatedAt?: string;
    public operationId?: string;
    public links?: Link[];

    public constructor(
        firstName: string, lastName: string, code: string, phoneNumber: string, email: string, address: Address
    ) {
        super(firstName, lastName, code, phoneNumber, email, address);
        this.id = Crypto.randomUUID();
        this.links = [];
    }
};

export type CourierCrud = z.infer<typeof schemas.CourierCrud>;

export type CourierDto = z.infer<typeof schemas.CourierDto>;

File: sources/core/Ggio.BikeSherpa.Backend.Domain/CourierAggregate/Courier.cs
using Ggio.DddCore;

namespace Ggio.BikeSherpa.Backend.Domain.CourierAggregate;

public class Courier : EntityBase<Guid>, IAggregateRoot, IAuditEntity
{
     public required string FirstName { get; set; }
     public required string LastName { get; set; }
     public required string Code { get; set; }
     public string? Email { get; set; }
     public required string PhoneNumber { get; set; }
     public required Address Address { get; set; }
     public DateTimeOffset CreatedAt { get; set; }
     public DateTimeOffset UpdatedAt { get; set; }
}